src/renderer/mod.rs

```diff
@@ -360,6 +364,12 @@ impl Renderer {
 
         root_canvas.restore();
 
+        let font_use_italic_as_oblique = { SETTINGS.get::<FontSettings>().use_italic_as_oblique };
```

Comment:
I really don't like idea of checking every frame for option change. There must be better way, @Kethku am i right?

I already see DrawCommand::FontChanged, which is sent on guifont change, maybe similar approach can be used for this?

Reply:
I don't like it also, to be honestly, I really tried to do another approach trying to do a single redraw request only when the setting is changed, but after many many tries through Editor and Renderer, I just couldn't find a way using what already exists

Reply:
If we couldn't find better way, i'd suggest putting this into separate function, like `is_font_options_changed`.

But let's wait for ideas from @Kethku , and then i'll try to think something.

Reply:
Another idea to think about, this is actually `font option`, we're already parsing guifont setting, and i found this in `:help guifont`

```
	For the Win32 GUI					*E244* *E245*
	- takes these options in the font name:
		hXX - height is XX (points, can be floating-point)
		wXX - width is XX (points, can be floating-point)
		b   - bold
		i   - italic
		u   - underline
		s   - strikeout
		cXX - character set XX.  Valid charsets are: ANSI, ARABIC,
		      BALTIC, CHINESEBIG5, DEFAULT, EASTEUROPE, GB2312, GREEK,
		      HANGEUL, HEBREW, JOHAB, MAC, OEM, RUSSIAN, SHIFTJIS,
		      SYMBOL, THAI, TURKISH, VIETNAMESE ANSI and BALTIC.
		      Normally you would use "cDEFAULT".

	  Use a ':' to separate the options.
```

So, those font options is gui-dependent, and we could try to use something like `o` to stand for oblique instead of italic.

Take a look at FontOptions::parse, i think this is exact place where this option should be defined, as it will allow us to drop this new FontSettings, and we already have code for detectiong changes in guifont.

Reply:
> Another idea to think about, this is actually `font option`, we're already parsing guifont setting, and i found this in `:help guifont`
> 
> ```
> 	For the Win32 GUI					*E244* *E245*
> 	- takes these options in the font name:
> 		hXX - height is XX (points, can be floating-point)
> 		wXX - width is XX (points, can be floating-point)
> 		b   - bold
> 		i   - italic
> 		u   - underline
> 		s   - strikeout
> 		cXX - character set XX.  Valid charsets are: ANSI, ARABIC,
> 		      BALTIC, CHINESEBIG5, DEFAULT, EASTEUROPE, GB2312, GREEK,
> 		      HANGEUL, HEBREW, JOHAB, MAC, OEM, RUSSIAN, SHIFTJIS,
> 		      SYMBOL, THAI, TURKISH, VIETNAMESE ANSI and BALTIC.
> 		      Normally you would use "cDEFAULT".
> 
> 	  Use a ':' to separate the options.
> ```
> 
> So, those font options is gui-dependent, and we could try to use something like `o` to stand for oblique instead of italic.
> 
> Take a look at FontOptions::parse, i think this is exact place where this option should be defined, as it will allow us to drop this new FontSettings, and we already have code for detectiong changes in guifont.

that should be ideal, I think, I didn't know that we could "break" compatibility with neovim gui options
and maybe I've misinterpreted that this options are to explicit state to use a bold at everywhere and not to use the font when a bold style occurs, it should be the right place to have a slant option (to italic or oblique) for sure

Reply:
Hmm, you're right, this option is about explicit font choice, not "when bold occurs".

Reply:
Renderer.draw_frame is called by GlutinWindowWrapper.draw_frame, and when font_changed, it calls `handle_new_grid_size`.

Does changing this option requires grid resize?

Reply:
Definitely not, it should only triggers a redraw at every window, just to update sections to use new style (and reusing blob cache at CachingShaper` to avoid reshaping unaffected parts)

I think, maybe, shouldn't it be triggered at `Editor` and call a `window.redraw()` at every window?
Just as guifont does when it's changed, because this is the exactly behavior which should be achieved

I don't really know how to triggers a redraw within `Renderer` itself in a way that doesn't checks every frame, maybe it could be done if `FontSettings` could implement `SettingGroup` to send a redraw command to renderer when it's value is updated, but maybe it's too far to just a single style option

Reply:
Problem is changing the guifont changes the grid sizing. I believe we currently try to preserve the window size as best as possible which triggers a grid resize. In the past we hadn't triggered redraws on font change, and that resulted in fonts being preserved between changes (big text in the command line, old font being kept around, etc)

As far as where to preserve these types of options, I have two points:
1. reading a font is just an index into a hashset to get a reference to the settings. This is VERY fast. So unless we see performance issues, the simplicity of just accessing the setting each time we draw is nice. If there is low hanging fruit to preserve the font change, then thats fine, but I wouldn't bend over backwards.
2. that said, moving the font setting into the guifont is a completely reasonable approach. As mentioned above, guifont is not a standard but rather specific to a given gui. If we feel the need to add an component of the guifont to better select and render fonts, I vote we just go for it.

Reply:
I feel using guifont to do that is the better approach to do that
but as we discussed, isn't guifont options stating that it should use that style everywhere?  (e.g use a font as bold everywhere as if it's a regular style)
at first, I've interpreted in this way, even if there is nothing at the docs saying otherwise (as far as I see)

but if it's just a misinterpretation of my side and that option states that is to use **when** a specific style occurs, I really agree with that, should be the best solution

Reply:
and if we all agree with that, I can make all needed modifications to fit into that

Reply:
I personally don't see an issue with guifont specifying that "when italic text is requested, render this variant". That feels like a top level font specification to me, but I also don't plan on using such a feature so I may be misunderstanding something

Reply:
Well, struct is already called `FontOptions`, so adding option like `prefers_oblique_for_italic` or `use_italic_as_oblique` feels right think to do.

And we can even make it private, and add `fn get_slant(self, italic: bool) -> Slant` for using in build_clusters.

Reply:
Right, I'll work on things to fit what was discussed